google-cloud-logging/lib/google/cloud/logging/middleware.rb

```diff
@@ -74,6 +74,56 @@ def extract_trace_id env
         end
 
         ##
+        # Construct a monitored resource based on the given type and label if
+        # both are provided. Otherwise, construct a default monitored resource
+        # based on the current environment.
+        #
+        # If not given both type and label:
+        #   If running from GAE, return resource:
+        #   {
+        #     type: "gae_app", {
+        #       module_id: [GAE module name],
+        #       version_id: [GAE module version]
+        #     }
+        #   }
+        #   If running from GKE, return resource:
+        #   {
+        #     type: "container", {
+        #       cluster_name: [GKE cluster name],
+        #       namespace_id: [GKE namespace_id]
+        #     }
+        #   }
+        #   If running from GCE, return resource:
+        #   {
+        #     type: "gce_instance", {
+        #       instance_id: [GCE VM instance id],
+        #       zone: [GCE vm group zone]
+        #     }
+        #   }
+        #   Otherwise default to { type: "global" }, which means not associated
+        #   with GCP.
```

Comment:
I originally thought using the JSON to represent return types makes it look prettier in source code. But after examining the generated yardoc, I agree your suggestion of change it to `@example` makes it look way prettier. I will make update in a sec.